apps/site/lib/site_web/plugs/banner_message.ex

```diff
@@ -0,0 +1,29 @@
+defmodule SiteWeb.Plugs.BannerMessage do
+  @moduledoc """
+  Assigns a message to `conn` using the following params:
+  - `message_key` is the key that will be used to retrieve the message.
+  - `message` is a map representing the actual message.
+  """
+  @behaviour Plug
+
+  import Plug.Conn, only: [assign: 3]
+
+  @impl true
+  def init(opts) do
+    opts
+  end
+
+  @impl true
+  def call(%Plug.Conn{assigns: %{tab: "trip-view"}} = conn, opts) do
```

Comment:
If this is going to be for general use we shouldn't limit it to the trip-view tab here. Also we'll want to make sure the message_key and message are both present before assigning so I think this might work better here:

```suggestion
  def call(conn, [message_key: message_key, message: message]) do
```